internal/cmd/connect/command_plugin_install.go

```diff
@@ -0,0 +1,458 @@
+	Name    string    `json:"name"`
+	Title   string    `json:"title"`
+	Version string    `json:"version"`
+	Owner   owner     `json:"owner"`
+	Archive archive   `json:"archive"`
+	Licenses []license `json:"license"`
+}
+
+type owner struct {
+	Username string `json:"username"`
+	Name     string `json:"name"`
+}
+
+type archive struct {
+	Url  string `json:"url"`
+	Md5  string `json:"md5"`
+	Sha1 string `json:"sha1"`
+}
+
+type license struct {
+	Name string `json:"name"`
+	Url  string `json:"url"`
+}
+
+const (
+	invalidDirectoryErrorMsg       = `plugin directory "%s" does not exist`
+	invalidWorkerConfigErrorMsg    = `worker config file "%s" does not exist`
+	unexpectedInstallationErrorMsg = "unexpected installation type: %s"
+)
+
+func (c *pluginCommand) newInstallCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "install <plugin>",
+		Short: "Install a Connect plugin.",
+		Args:  cobra.ExactArgs(1),
+		RunE:  c.install,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Install the latest release of the Datagen connector onto your local Confluent Platform environment.",
+				Code: "confluent connect plugin install confluentinc/kafka-connect-datagen:latest",
+			},
+			examples.Example{
+				Text: "Install the latest release of the Datagen connector in a user specified directory and update one worker-config file.",
+				Code: "confluent connect plugin install confluentinc/kafka-connect-datagen:latest --plugin-directory $CONFLUENT_HOME/plugins --worker-configs $CONFLUENT_HOME/etc/kafka/connect-distributed.properties",
+			},
+		),
+		Annotations: map[string]string{pcmd.RunRequirement: pcmd.RequireOnPremLogin},
+	}
+
+	cmd.Flags().String("plugin-directory", "", "The plugin installation directory. If not specified, a default will be selected based on your Confluent Platform installation.")
+	cmd.Flags().StringSlice("worker-configs", []string{}, "A comma-separated list of paths to one or more Kafka Connect worker configuration files. Each worker file will be updated to load plugins from the plugin directory in addition to any preexisting directories.")
+	pcmd.AddDryRunFlag(cmd)
+	cmd.Flags().Bool("force", false, "Proceed without user input.")
+
+	cobra.CheckErr(cmd.MarkFlagDirname("plugin-directory"))
+
+	return cmd
+}
+
+func (c *pluginCommand) install(cmd *cobra.Command, args []string) error {
+	dryRun, err := cmd.Flags().GetBool("dry-run")
+	if err != nil {
+		return err
+	}
+	if dryRun {
+		output.Println("[DRY RUN] Performing a dry run of this command.")
+	}
+
+	pluginManifest, err := getManifest(args[0])
+	if err != nil {
+		return err
+	}
+
+	force, err := cmd.Flags().GetBool("force")
+	if err != nil {
+		return err
+	}
+
+	// Parse plugin & worker flags
+	pluginDir, err := getPluginDirFromFlag(cmd)
+	if err != nil {
+		return err
+	}
+
+	workerConfigs, err := getWorkerConfigsFromFlag(cmd)
+	if err != nil {
+		return err
+	}
+
+	var installation *platformInstallation
+	prompt := form.NewPrompt(os.Stdin)
```

Comment:
Turns out the reason I couldn't provide multiple inputs was because I was creating a new prompt every time. Creating this one at the start and passing it around is the only way I found to allow integration testing.